server/src/main/java/org/candlepin/async/tasks/ImportJob.java

```diff
@@ -0,0 +1,226 @@
+
+            try {
+                JobArguments arguments = this.getJobArguments();
+
+                String ownerKey = arguments.getAsString(OWNER_KEY);
+                String storedFileId = arguments.getAsString(STORED_FILE_ID);
+                String uploadedFileName = arguments.getAsString(UPLOADED_FILE_NAME);
+
+                if (ownerKey == null || ownerKey.isEmpty()) {
+                    String errmsg = "owner has not been set, or the provided owner lacks a key";
+                    throw new JobConfigValidationException(errmsg);
+                }
+
+                if (storedFileId == null || storedFileId.isEmpty()) {
+                    String errmsg = "stored file id has not been set, or is empty";
+                    throw new JobConfigValidationException(errmsg);
+                }
+
+                if (uploadedFileName == null || uploadedFileName.isEmpty()) {
+                    String errmsg = "uploaded file name has not been set, or is empty";
+                    throw new JobConfigValidationException(errmsg);
+                }
+            }
+            catch (ArgumentConversionException e) {
+                String errmsg = "One or more required arguments are of the wrong type";
+                throw new JobConfigValidationException(errmsg, e);
+            }
+        }
+    }
+
+    private OwnerCurator ownerCurator;
+    private ManifestManager manifestManager;
+
+    @Inject
+    public ImportJob(OwnerCurator ownerCurator, ManifestManager manifestManager) {
+        this.ownerCurator = ownerCurator;
+        this.manifestManager = manifestManager;
+    }
+
+    @Override
+    public Object execute(JobExecutionContext context) throws JobExecutionException {
+        JobArguments args = context.getJobArguments();
+
+        String ownerKey = args.getAsString(OWNER_KEY);
+        ConflictOverrides overrides = new ConflictOverrides(args.getAs(CONFLICT_OVERRIDES, String[].class));
+        String storedFileId = args.getAsString(STORED_FILE_ID);
+        String uploadedFileName = args.getAsString(UPLOADED_FILE_NAME);
+
+        Throwable caught;
+        Owner targetOwner = null;
+        try {
+            targetOwner = ownerCurator.getByKey(ownerKey);
+            if (targetOwner == null) {
+                throw new NotFoundException(String.format("Owner %s was not found.", ownerKey));
+            }
+
+            ImportRecord importRecord = manifestManager.importStoredManifest(targetOwner,
+                storedFileId, overrides, uploadedFileName);
+            log.info("Async import complete.");
+            return importRecord;
+        }
+        // Catch and handle SyncDataFormatException and ImporterExceptions
+        // as the OwnerResource.importManifest does which will provide a little more
+        // info about the exception that was thrown (CandlepinException).
+        catch (SyncDataFormatException e) {
+            caught = new BadRequestException(e.getMessage(), e);
```

Comment:
I know this is how the Pinsetter version handled this, but is this what we want to continue doing? Something about the job system using exceptions intended for REST output gives me pause. Also, we have the terminal flag to juggle with some of these, so it should probably require a bit more attention (more on this below).

Reply:
I agree the REST codes do not really fit in a job failure output.
I'm happy to use your `JobFailureResult` gist, which can then be extended.
e.g.:
* Instead of `SyncDataFormatException`, we can use a `JobFailureResult` with appropriate type/message. Similar for `ImporterException`, and every exception jobs currently throw that do not have any extra data on them.
* Instead of `ImportConflictException`, we can extend `JobFailureResult` with a `private Set<Importer.Conflict> conflicts;`. Similarly for every other exception jobs currently throw that needs to have extra data on them?

Reply:
I guess what I'm getting at is that I'd rather exception handling be standardized, but even with the gist, I don't know how much value that will have. The dissonance we're seeing here is due to the special ExceptionMessage object CandlepinExceptions carry with them (which itself is kinda... eh... in my opinion). I'll take a quick glance at the other jobs; hopefully it's not something we're using elsewhere. And if that's the case, maybe the exception handling here should be updated to, at least, capture the CandlepinExceptions and disassembly manually, rather than propagating that up to the JobManager.

Reply:
I agree with this approach. I'll go with capturing CandlepinExceptions and propagating JobExecutionExceptions wrapping only the original non-CandlepinExceptions (e.g. SyncDataFormatException and ImporterException, but not BadRequestException or IseException). I've also commented on the other thread about details on how to deal with the unique nature of ImportConflictException,

Reply:
ack